controller-server/src/main/java/com/yahoo/vespa/hosted/controller/tenant/TenantInfo.java

```diff
@@ -0,0 +1,106 @@
+        this.name = Objects.requireNonNull(name);
+        this.email = Objects.requireNonNull(email);
+        this.website = Objects.requireNonNull(website);
+        this.contactName = Objects.requireNonNull(contactName);
+        this.contactEmail = Objects.requireNonNull(contactEmail);
+        this.invoiceEmail = Objects.requireNonNull(invoiceEmail);
+        this.address = Objects.requireNonNull(address);
+        this.billingContact = Objects.requireNonNull(billingContact);
+    }
+
+    public static TenantInfo EmptyInfo = new TenantInfo("","","", "", "", "",
+            TenantInfoAddress.EmptyAddress, TenantInfoBillingContact.EmptyBillingContact);
+
+    public String name() {
+        return name;
+    }
+
+    public String email() {
+        return email;
+    }
+
+    public String website() {
+        return website;
+    }
+
+    public String contactName() {
+        return contactName;
+    }
+
+    public String contactEmail() {
+        return contactEmail;
+    }
+
+    public String invoiceEmail() {
+        return invoiceEmail;
+    }
+
+    public TenantInfoAddress address() {
+        return address;
+    }
+
+    public TenantInfoBillingContact billingContact() {
+        return billingContact;
+    }
+
+    public TenantInfo withName(String newName) {
+        return new TenantInfo(newName, email, website, contactName, contactEmail, invoiceEmail, address, billingContact);
+    }
+
+    public TenantInfo withEmail(String newEmail) {
+        return new TenantInfo(name, newEmail, website, contactName, contactEmail, invoiceEmail, address, billingContact);
+    }
+
+    public TenantInfo withWebsite(String newWebsite) {
+        return new TenantInfo(name, email, newWebsite, contactName, contactEmail, invoiceEmail, address, billingContact);
+    }
+
+    public TenantInfo withContactName(String newContactName) {
+        return new TenantInfo(name, email, website, newContactName, contactEmail, invoiceEmail, address, billingContact);
+    }
+
+    public TenantInfo withContactEmail(String newContactEmail) {
+        return new TenantInfo(name, email, website, contactName, newContactEmail, invoiceEmail, address, billingContact);
+    }
+
+    public TenantInfo withInvoiceEmail(String newInvoiceEmail) {
+        return new TenantInfo(name, email, website, contactName, contactEmail, newInvoiceEmail, address, billingContact);
+    }
+
+    public TenantInfo withAddress(TenantInfoAddress newAddress) {
+        return new TenantInfo(name, email, website, contactName, contactEmail, invoiceEmail, newAddress, billingContact);
+    }
+
+    public TenantInfo withBillingContact(TenantInfoBillingContact newBillingContact) {
+        return new TenantInfo(name, email, website, contactName, contactEmail, invoiceEmail, address, newBillingContact);
+    }
+
+    public boolean isEmpty() {
+        return (name + email + website + contactEmail + contactName + invoiceEmail).isEmpty()
```

Comment:
If this implements equals/hashCode you can simply return `this.equals(EMPTY)`. Same for the other classes. This will also simplify serialization testing.

Reply:
Good catch!

Could you give a hint to what part of testing could be simplified due to this?

Reply:
If you add a fully populated `TenantInfo` to `@Test public void cloud_tenant()` you can do do `assertEquals(tenant.tenantInfo(), serialized.tenantInfo())` to test that serialization works as expected in both directions.

Reply:
Ah you where referring to if it implemented equals/hashCode - not how to implement the isEmpty method?I think that is exactly what is done right?

Reply:
Yes, never mind! You should still be able to simplify `isEmpty` though, by just comparing with empty.

Reply:
fixed